src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -132,6 +132,12 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
         }
       }
 
+      //Convert internal evalscript to V3 if it's not in that version.
+      if (this.evalscript && !this.evalscript.startsWith('//VERSION=3')) {
+        let evalscriptV3 = await this.convertEvalscriptToV3(this.evalscript);
```

Comment:
We are making an assumption here that the service will return an evalscript which starts with `'//VERSION=3'`. While this is true now, it is an undocumented feature. 

I know we did the same thing before and that this is not something you have changed, but I would suggest changing `convertEvalscriptToV3()` so that when evalscript is fetched, this condition is checked, and if false, the comment is prepended... Just to be on the safe side. If you feel this is out of scope feel free to resolve and merge.

Reply:
I am not sure whether this handled somehow, but what about basic evalscripts? Afaik evalscript converter *does not* convert basic evalscripts to full V3 format, and prepending `//VERSION=3` would result in an error on rendering.

Reply:
Good point! But this also means that the check `this.evalscript.startsWith('//VERSION=3')` is incomplete, since we are converting basic scripts every time, if they were already converted or not.  Not sure how to fix this though... Do we keep the original evalscript and put the converted one to `this.evalscriptV3`? Or do we set `this.evalscriptWasConvertedToV3` to true?

Reply:
I think it's almost certain that a script is basic of it doesn't contain `function evaluatePixel`, `function setup`, `setInputComponents(` etc. which would be possible to check, right? They have to be reserved names. After all, this is what service must be doing already.

Reply:
For what I'm checking, basic scripts don't get converted to V3 and they don't get prepended the //VERSION=3 comment.
Should we check if a script is basic in order to avoid doing unnecessary requests to the convertscript endpoint?

Reply:
I think it will be easier (and less error-prone) if we simply set `this.evalscriptWasConvertedToV3 = true;`) and then skip conversion if this flag is set. What do you think?

Reply:
Yes of course. I will do that. Thanks Anze.

Reply:
Can I merge it as it is right now Anze? @sinergise-anze

Reply:
Yes, absolutely! Looks good.